langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,568 @@
+                    "the Azure CosmosDB Mongo vCore connection string");
+        }
+
+        if (mongoClient == null) {
+            mongoClient = MongoClients.create(
+                    MongoClientSettings.builder()
+                            .applyConnectionString(new ConnectionString(endpoint))
+                            .applicationName(applicationName)
+                            .build());
+        }
+
+        MongoDatabase database = mongoClient.getDatabase(databaseName);
+        // create collection if not exist
+        if (!isCollectionExist(database, collectionName)) {
+            createCollection(database, collectionName, getOrDefault(createCollectionOptions, new CreateCollectionOptions()));
+        }
+        this.collection = database.getCollection(collectionName, AzureCosmosDbMongoVCoreDocument.class).withCodecRegistry(codecRegistry);
+
+        // create index if not exist
+        if (Boolean.TRUE.equals(createIndex) && !isIndexExist(this.indexName)) {
+            createIndex(this.indexName, collectionName, database);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = randomUUID();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(String id, Embedding embedding) {
+        addInternal(id, embedding, null);
+    }
+
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        List<Double> queryVector = referenceEmbedding.vectorAsList().stream()
+                .map(Float::doubleValue)
+                .collect(toList());
+
+        List<Bson> pipeline = new ArrayList<>();
+
+        switch (this.kind) {
+            case "vector-ivf":
+                pipeline = getPipelineDefinitionVectorIVF(queryVector, maxResults);
+                break;
+            case "vector-hnsw":
+                pipeline = getPipelineDefinitionVectorHNSW(queryVector, maxResults);
+                break;
```

Comment:
Created enum for this. And the enum has a method defined which will check if the passed string value matches the enum list, and if it doesn't we will set the default value which is vector-hnsw. We don't need a default case for this.

Reply:
If the value provided by the user is wrong, we should let him know about it (throw an exception) instead of silently "fixing" it which is very unobvious

Reply:
Added the check in the Enum itself.